ownCloud/FileLists/QueryFileListTableViewController.swift

```diff
@@ -444,46 +451,37 @@ class QueryFileListTableViewController: FileListTableViewController, SortBarDele
 		if !self.tableView.isEditing {
 			super.tableView(tableView, didSelectRowAt: indexPath)
 		} else {
-			updateMultiSelectionUI()
+			if let item = itemAt(indexPath: indexPath), let itemLocalID = item.localID {
+				selectedItemIds.append(itemLocalID as OCLocalID)
+				self.actionContext?.add(item: item)
+			}
+			updateMultipleSelectButtonItem()
+			updateActions()
 		}
 	}
 
 	override func tableView(_ tableView: UITableView, didDeselectRowAt indexPath: IndexPath) {
 		if tableView.isEditing {
-			updateMultiSelectionUI()
+			if let item = itemAt(indexPath: indexPath), let itemLocalID = item.localID {
+				selectedItemIds.removeAll(where: {$0.hash == itemLocalID.hash})
```

Comment:
The risk of hash collisions should be really low, yet I'm not sure comparing hashes is a shortcut that should be made.

Reply:
@felix-schwarz changed this.. I think I took this shortcut since compiler was failing to infer the type correctly.